### PR TITLE
N°7146 - Fix style not applied in list in the end-users portal in iTop 3.0+

### PR DESCRIPTION
**Pull request by accognet**
Add markup in order to use custom style  on status attributes

### CODE REVIEW

**Review by Molkobain — APPROVED** [review]

**Comment by Molkobain** [comment]
Support PRs review:
- Code factorized to match what is already done in the manage brick.
- Need to fix contact add before merge.

**Review by Molkobain — CHANGES_REQUESTED** [review]
Contact add doesn't work anymore, but we didn't find yet why.

**Comment by Molkobain** [comment]
Support PRs review: Contact add has been fixed, but some ext. keys lost their hyperlink. We will investigate further more.

**Comment by Molkobain** [comment]
Functional review: Accepted for 3.2
I'll add the tests to do in the ticket.

Technical: Discussed with @accognet , whilst style is displayed in a modal when picking elements to add to a linkedset, style is not displayed for the element already in the linkedset. She'll take a look, and depending on the timing, we'll merge what already works and rephrase the existing ticket to reflect that there is still a part to fix.

File: datamodels/2.x/itop-portal-base/portal/src/Controller/ObjectController.php [hunk 1]
```diff
@@ -1408,14 +1408,14 @@ public function GetInformationForLinkedSetAsJsonAction(Request $oRequest)
...
 		$sObjectClass = get_class($oObject);
-		$aObjectData = array(
-			'id'         => $oObject->GetKey(),
-			'name'       => $oObject->GetName(),
-			'attributes' => array(),
-		);
+		$aObjectData = [
+			'id'                  => $oObject->GetKey(),
+			'name'             => $oObject->GetName(),
+			'attributes'        => [],
```
Comment (Hipska) [inline datamodels/2.x/itop-portal-base/portal/src/Controller/ObjectController.php]: Don't know why these were added, before looked better:
```suggestion
			'id'         => $oObject->GetKey(),
			'name'       => $oObject->GetName(),
			'attributes' => [],
```

**Review by Hipska — CHANGES_REQUESTED** [review]

File: sources/Renderer/Bootstrap/FieldRenderer/BsLinkedSetFieldRenderer.php [hunk 2]
```diff
@@ -192,7 +193,7 @@ public function Render()
 								"render": function(data, type, row)
 								{
 									var oCheckboxElem = $('{$sSelectionInputHtml}');
-									if(row.limited_access)
+                                    if(row.limited_access)
```
Comment (Hipska) [inline sources/Renderer/Bootstrap/FieldRenderer/BsLinkedSetFieldRenderer.php:196]: It looks like your editor has wrong indentation settings configured..
```suggestion
									if(row.limited_access)
```

File: sources/Renderer/Bootstrap/FieldRenderer/BsLinkedSetFieldRenderer.php [hunk 3]
```diff
@@ -221,7 +222,8 @@ public function Render()
 							"className": {$sIsEditable} && aColumnProperties.mandatory ? 'mandatory' : '',
 							"render": function(data, type, row){
 								var cellElem;
-                                                                
+                                var metadataNames = ['object_class', 'object_id', 'attribute_code', 'attribute_type', 'value_raw'];
+
```
Comment (Hipska) [inline sources/Renderer/Bootstrap/FieldRenderer/BsLinkedSetFieldRenderer.php:226]:
```suggestion
								var metadataNames = ['object_class', 'object_id', 'attribute_code', 'attribute_type', 'value_raw'];

```

File: sources/Renderer/Bootstrap/FieldRenderer/BsLinkedSetFieldRenderer.php [hunk 4]
```diff
@@ -254,7 +266,8 @@ public function Render()
 							"data": "attributes." + sKey,
 							"className": aColumnProperties.mandatory ? 'mandatory' : '',
 							"render": function(data, type, row){
-								var cellElem;
+								var cellElem;                                
+                                var metadataNames = ['object_class', 'object_id', 'attribute_code', 'attribute_type', 'value_raw'];
```
Comment (Hipska) [inline sources/Renderer/Bootstrap/FieldRenderer/BsLinkedSetFieldRenderer.php:270]:
```suggestion
								var cellElem;
								var metadataNames = ['object_class', 'object_id', 'attribute_code', 'attribute_type', 'value_raw'];
```

**Review by Hipska — CHANGES_REQUESTED** [review]
Could you please check your IDE settings and replace all spaces with tabs?

**Review by Molkobain — APPROVED** [review]

**Pull request merged**